From def48f89c985736901bb7bbe427f65acd18ef8d6 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@oss.bmw-carit.de>
Date: Tue, 27 Sep 2016 15:25:19 +0200
Subject: [PATCH] address gcc6 build error

With gcc6, compiling fails with `stdlib.h: No such file or directory`,
as including '-isystem /usr/include' breaks with gcc6, cf.,
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70129.

This commit addresses this issue for this package in the same way
it was addressed in various other ROS packages. A list of related
commits and pull requests is at:

  https://github.com/ros/rosdistro/issues/12783

The SYSTEM attribute for the eigen and pcl include directories
was added in commit 9e876d2b [1] on 2012-09-16 during the
package's transition to ROS Groovy. The reason for using
the SYSTEM attribute cannot be inferred from that commit.

This attribute remained in the CMakeLists.txt during further
refinements in commits be4aebdb and 3a156140 on 2014-02-24, and
was not further touched until now.

[1] https://github.com/ros-planning/navigation/commit/9e876d2b45aca2e2166ac5a508b0e7c7abc50717
[2] https://github.com/ros-planning/navigation/commit/be4aebdb658381104b8e70f832091926e67ea0b4
[3] https://github.com/ros-planning/navigation/commit/3a156140439ea2300aaad45f7f2a65dab503ff24

Signed-off-by: Lukas Bulwahn <lukas.bulwahn@oss.bmw-carit.de>

Upstream-Status: Pending [https://github.com/ros-planning/navigation/pull/522]

Signed-off-by: Lukas Bulwahn <lukas.bulwahn@oss.bmw-carit.de>
---
 move_slow_and_clear/CMakeLists.txt | 5 ++---
 navfn/CMakeLists.txt               | 1 -
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/move_slow_and_clear/CMakeLists.txt b/move_slow_and_clear/CMakeLists.txt
index 078052d..686596e 100644
--- a/move_slow_and_clear/CMakeLists.txt
+++ b/move_slow_and_clear/CMakeLists.txt
@@ -19,9 +19,8 @@ find_package(Boost REQUIRED COMPONENTS thread)
 include_directories(
     include
     ${catkin_INCLUDE_DIRS}
-    SYSTEM
-        ${EIGEN_INCLUDE_DIRS}
-        ${PCL_INCLUDE_DIRS}
+    ${EIGEN_INCLUDE_DIRS}
+    ${PCL_INCLUDE_DIRS}
 )
 add_definitions(${EIGEN_DEFINITIONS})
 
diff --git a/navfn/CMakeLists.txt b/navfn/CMakeLists.txt
index ad01c3a..3e3995f 100644
--- a/navfn/CMakeLists.txt
+++ b/navfn/CMakeLists.txt
@@ -21,7 +21,6 @@ find_package(PCL REQUIRED)
 include_directories(
     include
     ${catkin_INCLUDE_DIRS}
-    SYSTEM
     ${EIGEN_INCLUDE_DIRS}
     ${PCL_INCLUDE_DIRS}
 )
-- 
2.5.5

